src/controllers/trip.comments.controller.js

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable curly */
+/* eslint-disable require-jsdoc */
+import tripCommentsServices from '../services/trip.comments.service';
+
+class tripCommentController {
+  static async createComment(req, res) {
+    try {
+      const { user } = req;
+      const data = {
+        ...req.body,
+        user_id: user.dataValues.id,
+        trip_id: req.trip.id
+      };
+      const sendComment = await tripCommentsServices.createComment(data);
+      if (!sendComment)
+        return res.status(400).json({ message: 'something went down' });
+      return res
+        .status(201)
+        .json({ message: 'comment sent successfully', sendComment });
+    } catch (err) {
+      return res.status(500).json({ message: 'internal server error' });
+    }
+  }
+
+  static async findAllComments(req, res) {
+    try {
+      const found = await tripCommentsServices.findTripComments();
+      if (!found)
+        return res.status(404).json({ message: 'comments does not exist' });
+      return res.status(200).json({ message: 'retrieved all comments', found });
+    } catch (err) {
+      return res.status(500).json({ message: 'internal server error' });
+    }
+  }
+
+  static async editedComment(req, res) {
+    try {
+      const { user } = req;
+      const foundComment = req.comment.dataValues;
+      const commentId = foundComment.id;
+      if (user.dataValues.id !== foundComment.user_id)
+        return res
+          .status(409)
+          .json({ message: 'you are not allowed to edit this comment' });
+      const commentUpdate = {
+        id: foundComment.id,
+        trip_id: foundComment.trip_id,
+        user_id: foundComment.user_id,
+        comment: req.body.comment ? req.body.comment : foundComment.comment
+      };
+      await tripCommentsServices.editComment(
+        { where: { id: commentId } },
+        commentUpdate
+      );
+      const findEdited = await tripCommentsServices.findSpecificComment(
+        commentId
+      );
+
+      return res
+        .status(200)
+        .json({ message: 'successfully edited a comment', findEdited });
+    } catch (err) {
+      return res.status(500).json({ message: 'internal server error' });
+    }
+  }
+
+  static async removeComment(req, res) {
+    try {
+      const { user } = req;
+      const foundComment = req.comment.dataValues;
+      const commentId = req.comment.dataValues.id;
+      if (user.dataValues.id !== foundComment.user_id)
+        return res
+          .status(409)
+          .json({ message: 'you are not allowed to delete this comment' });
+      const destroyComment = await tripCommentsServices.removeComment({
+        where: { id: commentId }
+      });
+      return res
+        .status(200)
+        .json({ message: 'successfuly removed a comment', destroyComment });
+    } catch (err) {
+      return res.status(500).json({ message: 'internal server error' });
+    }
```

Comment:
Let the developer know the error in all catch if unexpected

please do so in all the other catches.